larq/models.py

```diff
@@ -47,91 +67,316 @@ def _get_output_shape(layer):
+            ),
+            f"Memory\n({_bitsize_as_str(table_config['memory_units'])})",
+            *(
+                f"{i}-bit macs\n({_bitsize_as_str(table_config['mac_units'])})"
+                for i in table_config["mac_precisions"]
+            ),
         ]
-    ]
-    for layer, stats in zip(model.layers, layer_stats):
-        table.append(
+
+    def _generate_table_total(self, table_config):
+        row = ["Total", "", ""]
+        for i in table_config["param_bidtwidths"]:
+            row.append(
+                _format_table_entry(
+                    self.parameter_count(i), table_config["param_units"]
+                )
+            )
+        row.append(_format_table_entry(self.memory, table_config["memory_units"]))
+        for i in table_config["mac_precisions"]:
+            row.append(
+                _format_table_entry(self.op_count("mac", i), table_config["mac_units"])
+            )
+        return row
+
+    def generate_table(self, include_macs=True):
+        table_config = {
+            "param_bidtwidths": self.unique_param_bidtwidths,
+            "mac_precisions": self.unique_op_precisions if include_macs else [],
+            "param_units": 1,
+            "memory_units": 8 * 1024,
+            "mac_units": 8 * 1024,
+        }
+
+        table = []
+
+        table.append(self._generate_table_header(table_config))
+
+        for lp in self.layer_profiles:
+            table.append(lp.generate_table_row(table_config))
+
+        table.append(self._generate_table_total(table_config))
+
+        return table
+
+    def generate_summary(self, include_macs=True):
+        summary = [
+            ["Total params", self.parameter_count()],
+            ["Trainable params", self.parameter_count(trainable=True)],
+            ["Non-trainable params", self.parameter_count(trainable=False)],
+            ["Model size:", f"{self.memory / (8*1024*1024):.2f} MB"],
             [
-                layer.name,
-                _get_input_precision(layer),
-                _get_output_shape(layer),
-                *_row_from_stats(stats, summed_stat),
-                _compute_memory(stats),
-            ]
-        )
-    table.append(
-        [
-            "Total",
-            "",
-            "",
-            *_row_from_stats(summed_stat, summed_stat),
-            _compute_memory(summed_stat),
+                "Float-32 Equivalent",
+                f"{self.fp_equivalent_memory / (8*1024*1024):.2f} MB",
+            ],
+            ["Compression Ratio of Memory", self.memory / self.fp_equivalent_memory],
         ]
-    )
-    return table
+
+        if include_macs:
+            binarization_ratio = self.op_count("mac", 1) / self.op_count(op_type="mac")
+            summary.extend(
+                [
+                    ["Number of MACs", self.op_count(op_type="mac")],
+                    ["Ratio of MACs that are binarized", f"{binarization_ratio:.4f}"],
```

Comment:
What do you think about
```suggestion
                    ["Ratio of binarized MACs", f"{binarization_ratio:.4f}"],
```